### PR TITLE
EVG-18525: Replace Antd popconfirm

### PR DESCRIPTION
**Pull request by sophstad**
EVG-18525

### Description

- Take 2 of https://github.com/evergreen-ci/spruce/pull/1725
- Use LeafyGreen's Tooltip as the basis of our Popconfirm
- Support controlled or uncontrolled popconfirm
- Fix [bug](https://mongodb.slack.com/archives/C866SR2LR/p1679068270722309) that caused revert by adding `e.stopPropagation()` to popconfirms used in the expandable volume and host tables

### Testing

- Add unit tests for controlled and uncontrolled popconfirm
- Update Cypress tests

### CODE REVIEW

File: src/constants/tokens.ts [hunk 1]
```diff
@@ -16,10 +16,10 @@ const zIndex = {
 
   // Set these values to 1 to utilize LeafyGreen's built-in stacking context
   modal: 1,
+  popover: 1,
```
[the remark this answers is not in the document]
Comment (sophstad) [reply src/constants/tokens.ts:19]: Nitpicky but a z-index value of 30 meant that the menu was not "under" the modal backdrop
image

File: src/components/PatchActionButtons/DisableTasks.tsx [hunk 2]
```diff
@@ -20,10 +19,8 @@ export const DisableTasks: React.VFC<Props> = ({
   patchId,
   refetchQueries = [],
 }) => {
-  const [active, setActive] = useState(false);
-  const menuItemRef = useRef<HTMLDivElement>(null);
-
   const dispatchToast = useToastContext();
```
[the remark this answers is not in the document]
Comment (sophstad) [reply src/components/PatchActionButtons/DisableTasks.tsx:21]: Great find ❤️

[the remark this answers is not in the document]
Comment (minnakt) [reply src/components/PatchActionButtons/DisableTasks.tsx:21]: so smoooth!

File: src/pages/task/taskTabs/buildBaronAndAnnotations/FileTicketButton.tsx [hunk 3]
```diff
@@ -49,23 +49,23 @@ const FileTicketButton: React.VFC<FileTicketProps> = ({
   return (
     <Container>
       <Popconfirm
```
Comment (minnakt) [inline src/pages/task/taskTabs/buildBaronAndAnnotations/FileTicketButton.tsx:51]: strangely this button isn't working for me anymore! I restarted my dev server to make sure 🤔

File: src/pages/task/taskTabs/buildBaronAndAnnotations/AnnotationTicketsTable/AnnotationTicketsTable.tsx [hunk 4]
```diff
@@ -74,18 +75,14 @@ const AnnotationTicketsTable: React.VFC<AnnotationTicketsProps> = ({
             condition: userCanModify,
             wrapper: (children: JSX.Element) => (
               <Popconfirm
```
Comment (minnakt) [inline src/pages/task/taskTabs/buildBaronAndAnnotations/AnnotationTicketsTable/AnnotationTicketsTable.tsx:77]: these buttons don't seem to be working for me either 😭

also nit. perhaps we should update the button text from `Move To Suspected Issues` to `Move to suspected issues`

File: src/pages/task/taskTabs/buildBaronAndAnnotations/FileTicketButton.tsx [hunk 5]
```diff
@@ -49,23 +49,23 @@ const FileTicketButton: React.VFC<FileTicketProps> = ({
...
-        title="Do you want to create a failure ticket for this task?"
+        align="right"
         onConfirm={onClickFile}
-        icon={null}
-        placement="right"
-        okText="File Ticket"
-        okButtonProps={{ loading: loadingFileJiraTicket }}
-        cancelButtonProps={{ disabled: loadingFileJiraTicket }}
+        confirmText="File Ticket"
```
Comment (minnakt) [inline src/pages/task/taskTabs/buildBaronAndAnnotations/FileTicketButton.tsx:54]: nit. change casing maybe?

```suggestion
        confirmText="File ticket"
```

**Pull request merged**